### ISSUE TITLE
NameError when case/match dictionaries

### ISSUE THREAD

**Issue by vincent-lg** [Nuitka/Nuitka#2881]
## Summary

Nuitka doesn't seem to appreciate when we use a match statement with a case containing a partial dictionary, capturing the rest with the `**` syntax. This results in a NameError.

## Nuitka and Python version

```
(pyenv) C:\Users\...>python -m nuitka --version
2.2.2
Commercial: None
Python: 3.10.11 (tags/v3.10.11:7d4cc5a, Apr  5 2023, 00:38:17) [MSC v.1929 64 bit (AMD64)]
Flavor: CPython Official
Executable: C:\Users\...\Scripts\python.exe
OS: Windows
Arch: x86_64
WindowsRelease: 11
Version C compiler: C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.26.28801\bin\Hostx64\x64\cl.exe (cl 14.2).
```

## Install and Nuitka usage

Nuitka was installed in a virtual environment, using pip (`pip install nuitka`). No additional packages were present in the virtual environment.

## Steps to reproduce

1.  Write a small Python script:

    ```python
    dat = {"key1": 5, "key2": 312}
    match dat:
        case {"key1": 5, **rest}:
            print(f"Right dictionary: {rest}")
    ```

2.  Make sure it runs with Python (`python test.py`). It should print `Right dictionary: {'key2': 312}`.
3.  Use nuitka to compile the Python script (`nuitka test.py`). That doesn't generate unusual warnings that I can see:

    ```
    Nuitka-Options: Used command line options: test.py
    Nuitka-Options:WARNING: You did not specify to follow or include anything but main program. Check options and make sure
    Nuitka-Options:WARNING: that is intended.
    Nuitka: Starting Python compilation with Nuitka '2.2.2' on Python '3.10' commercial grade 'not installed'.
    Nuitka: Completed Python level compilation and optimization.
    Nuitka: Generating source code for C backend compiler.
    Nuitka: Running data composer tool for optimal constant value handling.
    Nuitka: Running C compilation via Scons.
    Nuitka-Scons: Backend C compiler: cl (cl 14.2).
    Nuitka-Scons: Backend linking program with 6 files (no progress information available for this stage).
    Nuitka-Scons: Compiled 6 C files using clcache with 3 cache hits and 3 cache misses.
    Nuitka: Keeping build directory 'test.build'.
    Nuitka: Successfully created 'test.exe'.
    Nuitka: Execute it by launching 'test.cmd', the batch file needs to set environment.
    ```

4.  Start the generated script:

    ```
    (pyenv) C:\Users...>test.cmd
    Traceback (most recent call last):
      File "C:\Users\...\test.py", line 4, in <module>
        print(f"Right dictionary: {rest}")
    NameError: name 'rest' is not defined

I have tried these simple steps using different Nuitka or Python versions (but always using a virtualenv). The result seems to be the same. Is it a limitation of the match statement? Other match operations seems to work just fine though.

Thanks for your great work,

**Comment by kayhayen** [comment]
The `.rest` value of `ast.MatchValue` has gone unimplemented it seems so far entirely.

**Comment by kayhayen** [comment]
Thanks for your report, this is fixed on the factory branch, which is a development version under rapid development. You can try it out by going here: https://nuitka.net/doc/factory.html 

Feedback on whether this is working is very welcome. Please do not share plans to do it; only confirm or deny that it is working.

**Comment by vincent-lg** [comment]
Thank you. I just tested in factory. All works well. You may close this issue. Thanks again,

**Comment by kayhayen** [comment]
I would normally make a hotfix out of it, since I take compatibility most serious, but this will be on 2.3 release and is on develop now.

**Comment by kayhayen** [comment]
The 2.3 release is due any day now.

**Comment by kayhayen** [comment]
I actually found more issues since I fixed this, and an error in how I dit it, could crash now when no key was given, but especially matching against `defaultdict` was also mutating values. All of that is fixed in 2.3 release.
